File: Leetcoding-Actions/Explore-Learn/Arrays/replaceElementWithGreatestElementOnRightSide.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        '''
        T: O(n) and S: O(1)
        '''
        currMax = -1
        for i in range(len(arr) - 1, -1, -1):
            currVal = arr[i]
            arr[i] = currMax
            currMax = max(currMax, currVal)
        return arr

File: Leetcoding-Actions/Explore-Learn/Arrays/replaceElementWithGreatestElementOnRightSide.java
class Solution {
    public int[] replaceElements(int[] arr) {
        /**
        * T: O(n) and O(1)
        */
        int currMax = -1;
        for (int i = arr.length - 1; i >= 0; i--) {
            int currVal = arr[i];
            arr[i] = currMax;
            currMax = Math.max(currMax, currVal);
        }
        return arr;
    }
}
